### DISCUSSION TITLE
Help: Self Referential Create

### DISCUSSION THREAD

**Discussion by ThePaulMcBride** [prisma/prisma#14196] [Q&A]
I have a model that looks something like this

```prisma
model Book {
  id          String   @id @default(uuid())
  title       String
  page_order  String[]
  pages       Page[]
}

model Page {
  id       String  @id @default(uuid())
  content  Json?
  book_id  String
  book     Book     @relation(fields: [book_id], references: [id])
}
```

I want to create a book, it's first page, and store that first page_id in the books page_order array. Is there a way to do this in a single update?

This is what I am currently doing, but is obviously doesn't add the page_id to the page_order field. I can run a second update to add it, but I would like to be able to do it in one update. Is there a way to do this?

```js
const book = await db.book.create({
  data: {
    account_id,
    title,
    pages: {
      create: [{ title: "Page 1" }],
    },
  },
});
```

**Comment by austincrim** [comment]
Hey @ThePaulMcBride 👋,

It isn't possible to do this in a single query with Prisma.

I would recommend looking at [interactive transactions](https://www.prisma.io/docs/concepts/components/prisma-client/transactions#interactive-transactions-in-preview) to accomplish this, doing both operations in a single transaction to ensure data accuracy.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
